### PR TITLE
Additional Convenience methods

### PR DESCRIPTION
**Pull request by xBalbinus**
Adding more convenience methods for #207

### CODE REVIEW

**Review by preston-evans98 — CHANGES_REQUESTED** [review]
Thanks for the PR @xBalbinus - looks great overall! If we can change the hardcoded QueryMode, I think this will be ready to merge

**Comment by preston-evans98** [comment]
Looks like this also needs a `cargo fmt` - you can use the included `makefile` to handle that and all of our other lints for you

File: db/sovereign-db/src/ledger_db/rpc.rs [hunk 1]
```diff
@@ -172,6 +172,59 @@ impl LedgerRpcProvider for LedgerDB {
...
+
+    fn get_batch_by_number(&self, number: u64) -> Result<Option<Self::BatchResponse>, anyhow::Error> {
+        self.get_batches(&[BatchIdentifier::Number(number)], QueryMode::Compact)
+            .map(|mut slots| slots.pop().unwrap_or(None))
+    }
+
+    fn get_tx_by_number(&self, number: u64) -> Result<Option<Self::TxResponse>, anyhow::Error> {
+        self.get_transactions(&[TxIdentifier::Number(number)], QueryMode::Compact)
+            .map(|mut txs| txs.pop().unwrap_or(None))
+    }
```
[the remark this answers is not in the document]
Comment (xBalbinus) [reply db/sovereign-db/src/ledger_db/rpc.rs]: 👍

**Comment by xBalbinus** [comment]
Addressed the above :) Thanks a ton for the review, any suggestions on which issue to tackle next are appreciated

**Review by preston-evans98 — APPROVED** [review]

**Comment by preston-evans98** [comment]
> Addressed the above :) Thanks a ton for the review, any suggestions on which issue to tackle next are appreciated

Feel free to pick anything from the open issues, but #191 is closely related here

**Pull request merged**